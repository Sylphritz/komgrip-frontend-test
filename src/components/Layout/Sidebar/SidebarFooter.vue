<template>
  <div class="sidenav-footer mx-3">
    <div class="card card-background shadow-none card-background-mask-secondary" id="sidenavCard">
      <div class="full-background" style="background-image: url('/images/white-curved.jpeg')"></div>
      <div class="card-body text-start p-3 w-100">
        <div class="docs-info">
          <h6 class="text-white up mb-0">Need help?</h6>
          <p class="text-xs font-weight-bold">CoinCap API 2.0</p>
          <a
            href="https://komgrip.co.th/coincap/"
            target="_blank"
            class="btn btn-white btn-sm w-100 mb-0"
            >Documentation</a
          >
        </div>
      </div>
    </div>
  </div>
</template>
