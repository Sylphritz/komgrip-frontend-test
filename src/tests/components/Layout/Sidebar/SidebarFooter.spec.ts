import { describe, it, expect } from 'vitest'

import { mount } from '@vue/test-utils'
import SidebarFooter from '@/components/Layout/Sidebar/SidebarFooter.vue'

describe('SidebarFooter', () => {
  it('matches the snapshot', () => {
    const wrapper = mount(SidebarFooter)
    expect(wrapper.element).toMatchSnapshot()
  })

  it('links to the API documentation', () => {
    const wrapper = mount(SidebarFooter)
    expect(wrapper.find('a').attributes('href')).toBe('https://komgrip.co.th/coincap/')
  })

  it('has the correct background image', () => {
    const wrapper = mount(SidebarFooter)
    expect(wrapper.find('.full-background').attributes('style')).toBe(
      "background-image:url('/images/white-curved.jpeg');"
    )
  })
})
