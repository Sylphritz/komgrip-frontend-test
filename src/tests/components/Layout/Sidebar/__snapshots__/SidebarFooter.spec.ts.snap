// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SidebarFooter > matches the snapshot 1`] = `
<div
  class="sidenav-footer mx-3"
>
  <div
    class="card card-background shadow-none card-background-mask-secondary"
    id="sidenavCard"
  >
    <div
      class="full-background"
      style="background-image:url('/images/white-curved.jpeg');"
    />
    <div
      class="card-body text-start p-3 w-100"
    >
      <div
        class="docs-info"
      >
        <h6
          class="text-white up mb-0"
        >
          Need help?
        </h6>
        <p
          class="text-xs font-weight-bold"
        >
          CoinCap API 2.0
        </p>
        <a
          class="btn btn-white btn-sm w-100 mb-0"
          href="https://komgrip.co.th/coincap/"
          target="_blank"
        >
          Documentation
        </a>
      </div>
    </div>
  </div>
</div>
`;
